gcovr/gcov_parser.py

```diff
@@ -954,86 +967,155 @@ def line_after_start(self, lineno: int, start: str, start_lineno: str) -> None:
         )
 
 
+def _process_exclusion_marker(
+    code: str,
+    lineno: int,
+    exlcude_prefix: str,
+    exclude_word: str,
+    exclude_postfix: str,
+    exclude_by_pattern_regex: Pattern[str],
+    warnings: _ExclusionRangeWarnings,
+    exclude_ranges: List[Tuple[int, int]],
+    exclusion_stack: List[Tuple[int, int]],
+) -> Tuple[List[Tuple[int, int]], List[Tuple[int, int]]]:
+    """
+    Process the exclusion marker.
+
+    Header is a marker name like LCOV or GCOVR.
+
+    START flags are added to the exlusion stack
+    STOP flags remove a marker from the exclusion stack
+    """
+
+    if _EXCLUDE_FLAG in code:
+
+        # marker exclusion
+        excl_pattern = re.compile(
+            "(" + exlcude_prefix + ")" + _EXCLUDE_FLAG + exclude_word + exclude_postfix
+        )
+        for header, flag in excl_pattern.findall(code):
+
+            if flag == "LINE":
+                if exclusion_stack:
+                    warnings.line_after_start(
+                        lineno,
+                        f"{header}" + _EXCLUDE_FLAG + exclude_word + "LINE",
+                        exclusion_stack[-1][1],
+                    )
+                else:
+                    exclude_ranges.append((lineno, lineno + 1))
+
+            if flag == "START":
+                exclusion_stack.append((header, lineno))
+
+            elif flag == "STOP":
+                if not exclusion_stack:
+                    warnings.stop_without_start(
+                        lineno,
+                        f"{header}" + _EXCLUDE_FLAG + exclude_word + "START",
+                        f"{header}" + _EXCLUDE_FLAG + exclude_word + "STOP",
+                    )
+                    continue
+
+                start_header, start_lineno = exclusion_stack.pop()
+                if header != start_header:
+                    warnings.mismatched_start_stop(
+                        start_lineno,
+                        f"{start_header}" + _EXCLUDE_FLAG + exclude_word + "START",
+                        lineno,
+                        f"{header}" + _EXCLUDE_FLAG + exclude_word + "STOP",
+                    )
+
+                exclude_ranges.append((start_lineno, lineno))
+
+            else:  # pragma: no cover
+                pass
+
+    if exclude_by_pattern_regex:
+        if exclude_by_pattern_regex.match(code):
+            exclude_ranges.append((lineno, lineno + 1))
+
+    return (
+        exclude_ranges,
+        exclusion_stack,
+    )
```

Comment:
Don't need to return this. This lists are only modified by append and therefore the modification is visible outside of the function.

Reply:
Thanks for the feedback, fixed